modules/tensor_mechanics/include/materials/ComputeLagrangianStressStVenantKirchhoff.h

```diff
@@ -0,0 +1,34 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ComputeLagrangianStressPK.h"
+
+/// St. Venant-Kirchhoff hyperelasticity
+//
+//    St. Venant-Kirchhoff hyperelasticity derivative from the
+//    strain energy function W = lambda / 2 tr(E)^2 + mu tr(E^2)
+//    with E = 1/2*(F^T*F - I)
+//
+class ComputeLagrangianStressStVenantKirchhoff : public ComputeLagrangianStressPK
+{
+public:
+  static InputParameters validParams();
+  ComputeLagrangianStressStVenantKirchhoff(const InputParameters & parameters);
+  virtual ~ComputeLagrangianStressStVenantKirchhoff(){};
+
+protected:
+  /// Actual stress/Jacobian update
+  virtual void computeQpPKStress();
+
+protected:
+  const Real _mu;
+  const Real _lambda;
```

Comment:
I think it is equivalent to use C*E as the constitutive law for SVK, am I right? That way users can take advantage of the `ComputeIsotropicElasticityTensor` and similar classes to define the material constants using any Lame pair, not just `lambda` and `mu`.

Reply:
I debated this myself.  Technically it's not a hyperelastic formulation if you use anything except an isotropic elasticity tensor.  But then you could define that tensor with any pairs of constants.  I'll move it over to use an elasticity tensor, but we should technically warn people if they don't elect an isotropic tensor.

Reply:
We have a Guarantee system for that exact purpose. In this case, the isotropic material is a GuaranteeConsumer requesting an isotropic elasticity tensor.

Reply:
Ah, okay.  I'll set it up this way.

Reply:
@hugary1995  Actually the following doesn't seem to trigger an error for a non-isotropic tensor:

`void
StVenantKirchhoffStressUpdate::initialSetup()
{
  // Enforce isotropic elastic tensor
  if (!hasGuaranteedMaterialProperty(_elasticity_tensor_name, Guarantee::ISOTROPIC))
    mooseError("StVenantKirchhoffStressUpdate requires an isotropic elasticity "
               "tensor");
}`

Any suggestion on what I'm doing wrong?

Reply:
@hugary1995 Nevermind, I should check the random snippets I grab from the docs.  I was using

'''
  [./elastic_tensor]
    type = ComputeElasticityTensor
    C_ijkl = '1022726 113636 113636 1022726 454545'
    fill_method = axisymmetric_rz
  [../]
'''

to test, which apparently just happens to use the right orthotropic values to give an isotropic tensor...